provisioner/template.go

```diff
@@ -229,19 +229,23 @@ func autoscalingBufferSettings(clusterOrData interface{}) (*podResources, error)
 				return nil, err
 			}
 
-			if instanceInfo.VCPU > currentLargestInstance.VCPU && instanceInfo.Memory > currentLargestInstance.Memory {
-				currentLargestInstance = instanceInfo
-			} else if instanceInfo.VCPU <= currentLargestInstance.VCPU && instanceInfo.Memory <= currentLargestInstance.Memory {
-				// do nothing
-			} else {
-				return nil, fmt.Errorf("unable to select autoscaling buffer settings, conflicting instance types %s and %s", currentLargestInstance.InstanceType, instanceInfo.InstanceType)
+			if instanceInfo.VCPU > currentLargestVCPU {
+				currentLargestVCPU = instanceInfo.VCPU
+			}
+
+			if instanceInfo.Memory > currentLargestMemory {
+				currentLargestMemory = instanceInfo.Memory
 			}
 		}
 	}
 
+	if currentLargestVCPU == 0 || currentLargestMemory == 0 {
+		return nil, fmt.Errorf("unable to select autoscaling buffer settings, invalid CPU setting %d and Memory setting %d", currentLargestVCPU, currentLargestMemory)
+	}
+
 	result := &podResources{
-		CPU:    k8sresource.NewMilliQuantity(effectiveQuantity(currentLargestInstance.VCPU*1000, cpuScale, cpuReserved), k8sresource.DecimalSI).String(),
-		Memory: k8sresource.NewQuantity(effectiveQuantity(currentLargestInstance.Memory, memoryScale, memoryReserved), k8sresource.BinarySI).String(),
+		CPU:    k8sresource.NewMilliQuantity(effectiveQuantity(currentLargestVCPU*1000, cpuScale, cpuReserved), k8sresource.DecimalSI).String(),
```

Comment:
This should now compute the smallest available vCPU/memory instead, I think. The intent originally was to handle situations where you have node type combinations where some nodes have more CPU but less memory than the others. Previously this resulted in an error, and this code will configure autoscaling buffers that can't be scheduled at all.